clients/vault/src/execution.rs

```diff
@@ -259,6 +259,42 @@ impl Request {
 	}
 }
 
+/// executes open request based on the transaction
+async fn _execute_open_requests(
+	transaction: TransactionResponse,
+	oracle_agent: Arc<OracleAgent>,
+	parachain_rpc: SpacewalkParachain,
+	request: Request,
+) {
+	// max of 3 retries for failed request execution
+	let max_retries = 3;
+	let mut retry_count = 0;
+
+	if let Ok(tx_env) = transaction.to_envelope() {
+		let slot = transaction.ledger as Slot;
+		while retry_count < max_retries {
+			match oracle_agent.get_proof(slot).await {
+				Ok(proof) => {
+					if let Err(e) =
+						request.execute(parachain_rpc.clone(), tx_env.clone(), proof).await
+					{
+						tracing::error!("Failed to execute request #{}: {}", request.hash, e);
+						retry_count += 1; // increase retry count
+						continue
+					}
+					break
+				},
+				Err(error) => {
+					retry_count += 1; // increase retry count
+					tracing::error!("Failed to get proof for slot {}: {:?}", slot, error);
+				},
+			}
+		}
```

Comment:
Should we add a line that prints some log if we exhausted all our retry counts and it still did not work? Or should we maybe in l. 280 mention that we are retrying and also mention the retry count?